### DISCUSSION TITLE
Case of a transient thermal analysis with forced convection BC

### DISCUSSION THREAD

**Discussion by yvanblanchard** [adtzlr/felupe#860] [General]
Hello
Before exploring feasibility with PetSc library, I wanted to check first if Felupe lib (much more easier to use) would be able to model this case:
I have a rotated rubber roller (which is part of the example test cases here) but which was heated and then cooled by a forced convection (air cooling using a kind of torch duct). This it is a transient heat transfer analysis (with 2D model of roller section, using rubber material) and the fact that roller is moving also contributes to accelerate the cooling (only a side of roller is cooled by fixed/static duct device).
Is it possible to model ?
Thank you

**Comment by adtzlr** [comment]
Hi @yvanblanchard,

as you may have expected - unfortunately no. At least there are no high-level functions/classes. If you know your equations, you can implement them using the weak form expressions [`felupe.Form`](https://felupe.readthedocs.io/en/latest/felupe/assembly.html#felupe.Form).

**Comment by yvanblanchard** [comment]
Thank you @adtzlr

**Comment by yvanblanchard** [comment]
Hello
I would like to know if a much easier test case could modelled with Felupe : hollowed cylinder with external flux and internal convection (cooling), using either 2D disk mesh or à rectangular shape mesh (wall section of cylinder) with axisymetric conditions.

thank you!

<img width="418" alt="image" src="https://github.com/user-attachments/assets/334744e9-c40c-4640-9d19-7b204d672252">

**Reply by adtzlr** [reply]
Hey @yvanblanchard,

sorry for not answering your question. This should be possible now, are you still interested?

**Comment by yvanblanchard** [comment]
Hi @adtzlr 
Yes still interested , thanks for feedback.
It would be a structure made of  inner Metallic shaft with an envelope of hyperelastic rubber (no more hollowed tube), if possible. It is heated by an external flux (hot gas or laser) at a side, and cooled by pulsed air (either internally or through a nozzle externally at other side).
I tried to model it with Calculix solver but not so easy.
Thanks!

**Reply by adtzlr** [reply]
Sorry, I meant the simpler model. Without thermal-mechanical coupling, just the thermal part. For axisymmetric thermal problems, we need to resolve #1043 first.
